### PR TITLE
build: Support arm64 container and add containerized build target

### PR DESCRIPTION
**Pull request by pingleig**
# Description of the issue

The Dockerfile is hard coded to use amd64 binary, we should determine it at runtime or let build context specify it.

# Description of changes

- Use `TARGETARCH` arg from `docker buildx` or fallback to `dpkg --print-architecture`
- Add two new Dockerfiles to build from local deb file or the source code
- Change base builder image from Debian to Ubuntu because there is no official release for Debian arm64

# License

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

# Tests

```bash
cd amazon-cloudwatch-container-insights/cloudwatch-agent-dockerfile
docker build .
```

```bash
# make sure you have access to go mod proxy,  some organization block it in corp network
make dockerized-build
# If you have the repo with vendor folder,  run the following instead, go mod proxy access is not required
make dockerized-build-vendor
```

For multi arch build, make sure you have docker buildx, you need to use the edge version instead of the stable one.

```bash
cd amazon-cloudwatch-container-insights/cloudwatch-agent-dockerfile
docker buildx create --name multi-builder
docker buildx use multi-builder
docker buildx build --platform linux/amd64,linux/arm64 .
```

### CODE REVIEW

File: amazon-cloudwatch-container-insights/cloudwatch-agent-dockerfile/README.md [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+Example using `manifest-tool`
+
+```bash
+# NOTE: the released version of manifest-tool is a bit outdated, you need to build it from source
+manifest-tool push from-spec multi-arch-agent.yaml
+```
+
+```yaml
+# multi-arch-agent.yaml
+image: 123.dkr.ecr.us-west-2.amazonaws.com/cloudwatch-agent:foo
```
Comment (pxaws) [inline amazon-cloudwatch-container-insights/cloudwatch-agent-dockerfile/README.md:50]: I think it is more clear if we use `account-id` to replace `123` and use `region` to replace `us-west-2`

File: amazon-cloudwatch-container-insights/cloudwatch-agent-dockerfile/README.md [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+You need to be aware of the following:
+
+- Single arch images should already exists on registry first because the multi arch image is reference to existing images on the registry.
+  - `docker buildx` is an exception because it pushes blob to registry without creating a new tag for the single arch images.
+- Both [docker manifest](https://docs.docker.com/engine/reference/commandline/manifest/) command and [manifest-tool](https://github.com/estesp/manifest-tool) should work, `manifest-tool` does not requires a docker daemon.
+
+Example using `docker manifest`
+
+```bash
+# NOTE: manifest is a experimental command, enable experimental in your ~/.docker/config.json with:
```
[the remark this answers is not in the document]
Comment (pingleig) [reply amazon-cloudwatch-container-insights/cloudwatch-agent-dockerfile/README.md]: There is no date in the official document, though based on the PR https://github.com/docker/cli/pull/138  docker release   built after Mid 2018 should have it.

File: amazon-cloudwatch-container-insights/cloudwatch-agent-dockerfile/README.md (hunk 1 above)
Comment (pingleig) [reply amazon-cloudwatch-container-insights/cloudwatch-agent-dockerfile/README.md]: Updated the template and mention it's for ECR

File: amazon-cloudwatch-container-insights/cloudwatch-agent-dockerfile/README.md [hunk 3]
```diff
@@ -0,0 +1,65 @@
+# CloudWatch Agent Dockerfiles
+
+- [Dockerfile](Dockerfile) builds from the [latest release published on s3](https://docs.aws.amazon.com/AmazonCloudWatch/latest/monitoring/install-CloudWatch-Agent-commandline-fleet.html)
+- [locadeb](localdeb/Dockerfile) builds from a local deb file
```
[the remark this answers is not in the document]
Comment (pingleig) [reply amazon-cloudwatch-container-insights/cloudwatch-agent-dockerfile/README.md]: done

**Review by pxaws — APPROVED** [review]

**Pull request merged**